src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/UsageBasedMetadataManager.cs

```diff
@@ -492,7 +481,7 @@ public override void GetDependenciesDueToAccess(ref DependencyList dependencies,
         public override void GetDependenciesDueToAccess(ref DependencyList dependencies, NodeFactory factory, MethodIL methodIL, MethodDesc calledMethod)
         {
             bool scanReflection = (_generationOptions & UsageBasedMetadataGenerationOptions.ReflectionILScanning) != 0;
-            if (scanReflection && FlowAnnotations.RequiresDataflowAnalysis(calledMethod))
+            if (scanReflection && Dataflow.ReflectionMethodBodyScanner.RequiresReflectionMethodBodyScannerForCallSite(FlowAnnotations, calledMethod))
```

Comment:
This was an ooops. We were running the analysis less often than we should (we need to run it not just when calling into an annotated method, but also if the method is an intrinsic. Most, but not all, intrinsics have annotations.